core/tee/tee_rpmb_fs.c

```diff
@@ -43,6 +43,15 @@
 
 #define TEE_RPMB_FS_FILENAME_LENGTH 224
 
+#if (defined CFG_RPMB_FS_CACHE_SIZE && CFG_RPMB_FS_CACHE_SIZE > 0)
```

Comment:
`defined CFG_RPMB_FS_CACHE_SIZE` is not needed. We can safely assume that `CFG_RPMB_FS_CACHE_SIZE` is always defined.
Is there no way of getting rid of all the `#ifdefs`? Why is zero such a special case compared to 1 or any other value?

Reply:
Hi @jforissier

- Given your answer, I see that the 'defined' and '!defined' statements can be removed (as we can safely assume 'CFG_RPMB_FS_CACHE_SIZE' is always defined)
- I can get rid of most/all '#ifdefs'. Here's my thoughts: Now, without having 'CFG_RPMB_FS_CACHE=y/n' and only having a single 'define', namely 'CFG_RPMB_FS_CACHE_SIZE', setting 'CFG_RPMB_FS_CACHE_SIZE=0' means that the caching feature is deactivated. In case the caching feature is activated, i.e., '(CFG_RPMB_FS_CACHE_SIZE>0)', we need to execute different logic than if caching would be deactivated in some places. The way I can get rid of the e.g. '#ifdef CFG_RPMB_FS_CACHE_SIZE > 0' would be to replace it with 'if (CFG_RPMB_FS_CACHE_SIZE > 0) { … } else { … }' statements in the code. We'd get rid of the 'ifdef's', but we'd have more 'if statements' in the code itself, leading to more indentation. Please let me know if you were referring to that, I'm of course happy to make any such change.

Manuel